### PR TITLE
Update tests, upgrade Python syntax, fix some flake8

### PR DESCRIPTION
**Pull request by hugovk**
* Upgrade unit tests to use more useful asserts
* Upgrade Python syntax with [pyupgrade](https://github.com/asottile/pyupgrade).
* Fix some flake8 warnings
* Fix some little typos

### CODE REVIEW

File: knack/help.py [hunk 1]
```diff
@@ -426,25 +426,25 @@ def _print_items(layouts):
 
     @staticmethod
     def _get_choices_defaults_sources_str(p):
-        choice_str = u'  Allowed values: {0}.'.format(', '.join(sorted([str(x) for x in p.choices]))) \
```
Comment (tjprescott) [inline knack/help.py]: Okay with removing the numbers in the curly braces, but the leading `u` needs to remain for now.

File: knack/invocation.py [hunk 2]
```diff
@@ -57,9 +57,9 @@ def __init__(self,
     def _filter_params(self, args):  # pylint: disable=no-self-use
         # Consider - we are using any args that start with an underscore (_) as 'private'
         # arguments and remove them from the arguments that we pass to the actual function.
-        params = dict([(key, value)
```
Comment (tjprescott) [inline knack/invocation.py]: Please pull this change into a separate PR.

File: knack/util.py [hunk 3]
```diff
@@ -73,8 +73,8 @@ def todict(obj, post_processor=None):  # pylint: disable=too-many-return-stateme
     elif hasattr(obj, '_asdict'):
         return todict(obj._asdict(), post_processor)
     elif hasattr(obj, '__dict__'):
-        result = dict([(to_camel_case(k), todict(v, post_processor))
```
Comment (tjprescott) [inline knack/util.py]: Please pull this change into a separate PR.

File: knack/invocation.py (hunk 2 above)
Comment (hugovk) [reply knack/invocation.py]: Done, please see https://github.com/Microsoft/knack/pull/107.

File: knack/help.py (hunk 1 above)
Comment (hugovk) [reply knack/help.py]: Done!
